src/Controller.js

```diff
@@ -0,0 +1,257 @@
+import Api from './Api';
+import Model from './Model';
+const $ = require( 'jquery' );
+/**
+ * An activation singleton, responsible for activating and attaching the
+ * button that activates the system when it is applicable.
+ *
+ * @class
+ */
+class Controller {
+	/**
+	 * Initialize if it is the first time, and cache the
+	 * initialization for the next times.
+	 */
+	constructor() {
+		if ( !Controller.instance ) {
+			this.initialized = false;
+			this.link = null;
+			this.namespace = null;
+			this.mainPage = false;
+			this.lang = 'en';
+			this.translations = {};
+
+			Controller.instance = this;
+
+			this.app = null;
+			this.api = null;
+			this.model = new Model();
+
+			// Events
+			this.model.on( 'active', isActive => {
+				this.toggleLinkActiveState( isActive );
+				// Toggle a class for CSS to style links appropriately.
+				this.model.getContentWrapper()
+					.toggleClass( 'wwt-active', isActive );
+			} );
+			// Events
+			this.model.on( 'enabled', isEnabled => {
+				this.getButton().toggle( isEnabled );
+			} );
+		}
+
+		return Controller.instance;
+	}
+
+	getModel() {
+		return this.model;
+	}
+
+	getApi() {
```

Comment:
Needs a docblock.

Reply:
egh we really need to add that to eslint